spec/haskell/src/SEL4/Kernel/Thread.lhs

```diff
@@ -352,12 +353,11 @@ An IPC operation may request that the scheduler switch to a specific thread.
 We check here that the candidate has the highest priority in the system.
 
 >         SwitchToThread candidate -> do
->             isSchedulable <- isSchedulable curThread
 >             when isSchedulable (tcbSchedEnqueue curThread)
 >
 >             idleThread <- getIdleThread
 >             targetPrio <- threadGet tcbPriority candidate
->             curPrio <- if idleThread == curThread then return 0 else threadGet tcbPriority curThread
+>             curPrio <- if (curThread /= idleThread) then (threadGet tcbPriority curThread) else (return 0)
```

Comment:
Just curious, did you have a specific reason for this change?

Reply:
It corresponds more closely to the other specifications.